mautic/mautic
# #5615 Fix - Multi-Select Fields Sync Not Working

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Yes
| New feature? | No 
| Automated tests included? | No
| Related user documentation PR URL |  N/A
| Related developer documentation PR URL | N/A 
| Issues addressed (#s or URLs) | #5615 
| BC breaks? | No
| Deprecations? | No

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description: 
Multi-Select fields between Mautic and SugarCRM were not properly syncing/displaying due to different formats. Mautic stores the multi-select string in format as `'choice1|choice2|choice_3'`, while SugarmCRM/SuiteCRM stores the multi-select string in format as `'^choice1^,^choice2^,^choice_3^'`. Added methods to check for and transform multi-select strings in the SugarCRM integration when pulling or pushing accounts, contacts or leads.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Make sure a SugarCRM 6.x integration is properly setup and syncing
2. Setup a multi-select field in SugarCRM, take careful note of the values inputted
3. Setup a multi-select field in Mautic making sure to use the exact same values inputted in SugarCRM
4. Open an Account, Contact or Lead record in SugarCRM (whichever you setup the multi-select in) and edit the contact to choose one or more choices in the multi-select field.
5. Trigger the `fetchleads console/cron job php app/console mautic:integration:fetchleads --integration=Sugarcrm --time-interval="1 minute"`
6. Wait for the sync to complete and then check the corresponding record in the Mautic UI. Observe that nothing is displayed.
7. View that same Mautic record in the Mautic database and observe the the field did sync and the value stored there, but in a format that Mautic does not understand.
8. Feel free to also try steps 4 - 7 reversing the systems (editing in Mautic and then checking in SugarCRM). Observe the same behavior either way.

#### Steps to test this PR:
1. Pull the pull request commit(s) into your local branch
2. Follow 'Steps to Reproduce the bug' above.
3. Observe that the synced multi-select values now show up in both Mautic and SugarCRM/SuiteCRM for Companies and Leads / (Accounts, Contacts and Leads).

#### List deprecations along with the new alternative:
N/A

#### List backwards compatibility breaks:
N/A
